aten/src/ATen/native/cuda/Indexing.cu

```diff
@@ -1350,7 +1350,7 @@ template <typename scalar_t>
 void index_select_out_cuda_impl(
     Tensor& out,
     const Tensor& self,
-    long dim,
+    uint64_t dim,
```

Comment:
Do we expect dim to be positive? Is negative indexing supported? Or do we desugar into a nonnegative int at this point?

Reply:
yeah it's desugared here: https://github.com/pytorch/pytorch/blob/50efbb9f1e7111b4b6d5b8e9a6064ee9783930be/aten/src/ATen/native/cuda/Indexing.cu#L1506

Reply:
Ah, but isn't the return type of at::maybe_wrap_dim `int64_t`? Wouldn't this warning just move up a level?

Reply:
I don't think that conversion necessarily causes a warning, it's the signed unsigned comparison?

Reply:
(added an explicit cast to avoid implicit conversion)